lib/portage/sync/syncbase.py

```diff
@@ -330,21 +330,23 @@ def noisy_refresh_keys():
 
     def _get_openpgp_env(self, openpgp_key_path=None, debug=False):
         if gemato is not None:
+            timeout = 15
             # Override global proxy setting with one provided in emerge configuration
             if "http_proxy" in self.spawn_kwargs["env"]:
                 proxy = self.spawn_kwargs["env"]["http_proxy"]
             elif "https_proxy" in self.spawn_kwargs["env"]:
                 proxy = self.spawn_kwargs["env"]["https_proxy"]
             else:
                 proxy = None
+                timeout = 5
 
             if openpgp_key_path:
                 openpgp_env = gemato.openpgp.OpenPGPEnvironment(
-                    proxy=proxy, debug=debug
+                    proxy=proxy, debug=debug, timeout=timeout
                 )
```

Comment:
Does this timeout mean that callers should be adjusted to handle `requests.Timeout` exceptions?

We could possibly catch it here and the re-raise it as `portage.exeception.TimeoutException` or something.

Reply:
Hmm, possibly. It doesn't look like gemato handles requests.Timeout nor uses the base exception classes.

Reply:
How do you want to handle this given that portage doesn't directly depend on requests?

Reply:
We can substitute TimeoutError if requests is missing like this:
```python
try:
	from requests.exceptions import Timeout
execept ImportError:
	Timeout = TimeoutError
```